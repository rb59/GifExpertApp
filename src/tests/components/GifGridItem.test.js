import React from 'react';
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe('Pruebas en GifGridItem', () => {

    const title = 'test';
    const url= 'https://imagen.test';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('debe renderizar el componente correctamente', () => {

        
        

        expect(wrapper).toMatchSnapshot();

    });

    test('debe tener un parrafo con el titulo', () => {

        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);
        
    });

    test('debe tener url y alt correctos', () => {

        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
        
    });

    test('debe tener la clase animate__fadeIn', () => {

        const div = wrapper.find('div');
        expect(div.hasClass('animate__fadeIn')).toBe(true);
        
    });
    
    
});
