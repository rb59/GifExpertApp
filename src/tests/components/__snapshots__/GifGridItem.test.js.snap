// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe renderizar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="test"
    src="https://imagen.test"
  />
  <p>
    test
  </p>
</div>
`;
